libusb/os/darwin_usb.c

```diff
@@ -1675,8 +1675,8 @@ static int submit_iso_transfer(struct usbi_transfer *itransfer) {
 
   struct darwin_interface *cInterface;
 
-  /* construct an array of IOUSBIsocFrames, reuse the old one if possible */
-  if (tpriv->isoc_framelist && tpriv->num_iso_packets != transfer->num_iso_packets) {
+  /* construct an array of IOUSBIsocFrames, reuse the old one if the sizes are the same */
+  if (tpriv->num_iso_packets != transfer->num_iso_packets) {
```

Comment:
This change should NOT be part of the same PR.

Reply:
Why? It's the same thing: removing a check for null before calling free().

Reply:
Exact. Sorry.